### ISSUE TITLE
Make it possible to interrupt KeyboardDragListener

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/scenery-phet#368]
From https://github.com/phetsims/scenery/issues/764, @jonathanolson said

>Also I see no way to interrupt one of those listeners.

There is interruption now on `blur`, but there is no public function to  do so. Ill work on this.

**Comment by jessegreenberg** [comment]
Visibility annotations in KeyboardDragListener are missing, I will add as part of this issue.

**Comment by jessegreenberg** [comment]
`reset` is actually the way to interrupt the keyboard dragging. It will reset the `keyState` and press and hold intervals and stop all dragging until the next input events. So maybe we should have a public `interrupt` function that just calls reset so that it is easily discoverable in the API?

**Comment by jessegreenberg** [comment]
Or maybe better to rename reset to interrupt. Ill do that.

**Comment by jessegreenberg** [comment]
@zepumph volunteered to work on this, thanks!

**Comment by jonathanolson** [comment]
`interrupt` would be more consistent with the usual input listeners. Additionally, maybe `Node.interruptInput` should go through a11y listeners and, if it exists, call `interrupt`?

**Comment by zepumph** [comment]
Progress was made on this. I basically did 3 things.

1. Consolidated `blur` and `reset`, all into a `interrupt` function. Those two function no longer exist.
2. Added an accessibility specific interrupt function, that is called by Node.
3. Added code to call `Node.interruptSubtreeInput()` from the a11y `blur()` function. I was easily able to interrupt a keyboardDragHandler (BASE/Friction) by touching my touch screen while keyboard dragging. This was the only way to accomplish this interrupt functionality.  @jonathanolson does hearing this make you worried about anything (just this third item)?

@jessegreenberg please review commits, anything else?

**Comment by jonathanolson** [comment]
> Added code to call Node.interruptSubtreeInput() from the a11y blur() function. I was easily able to interrupt a keyboardDragHandler (BASE/Friction) by touching my touch screen while keyboard dragging. This was the only way to accomplish this interrupt functionality. @jonathanolson does hearing this make you worried about anything (just this third item)?

When is blur() called? It also seems inconsistent to call interruptSubtreeInput only sometimes, AND to do so when we might be doing something that will immediately focus the node again.

**Comment by jessegreenberg** [comment]
That case seems very rare, and like it would be bad practice for a11y. I think that interrupting on blur is the right way to go.

**Comment by zepumph** [comment]
>It was added so that reset was called when the blur event was sent to the node's primary HTMLElement, sorry that wasn't clear. But @zepumph your changes handle this more generally.

Ahh I think I get it! A blur event on a DOMElement will call the blur event on all it's listeners? Or is there a place where `KeyboardDragHandler.blur` was being registered with the primary DOMElement's blur event? 

I'm asking so that I can make sure I understand the problem fully, and thus have implemented it completely.

**Comment by zepumph** [comment]
I ran into a bug where using a keyboard to press a button with a presslistener would throw an error because the presslistener expects a pointer, so I changed the blur function in the a11y trait to just interrupt a11y related listeners.

**Comment by jessegreenberg** [comment]
Sounds good. The only other thing I noticed was in `interruptAccessibleInput`, 
```js
var listenersCopy = this._accessibleInputListeners;
```
wasn't actually a copy. Changed in  https://github.com/phetsims/scenery/commit/18574a1a3354fa1f44be5c9c0eb3ea0410ca196e, otherwise I think everything looks good. @zepumph can this be closed?

**Comment by zepumph** [comment]
Sounds great. Thanks @jessegreenberg.
